packages/affiliates/libs/uniswap/contracts.ts

```diff
@@ -0,0 +1,121 @@
+import assert from "assert";
+import { ethers } from "ethers";
+import Promise from "bluebird";
+
+import UniswapV3Pool from "@uniswap/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json";
+
+import { Positions, Position, Pool } from "./models";
+import { PoolEvents } from "./processor";
+import { convertValuesToString } from "./utils";
+
+type NetworkName = "kovan" | "rinkeby" | "ropsten";
+type ContractType =
+  | "v3CoreFactoryAddress"
+  | "weth9Address"
+  | "multicall2Address"
+  | "proxyAdminAddress"
+  | "tickLensAddress"
+  | "quoterAddress"
+  | "swapRouter"
+  | "nonfungibleTokenPositionDescriptorAddress"
+  | "descriptorProxyAddress"
+  | "nonfungibleTokenPositionManagerAddress"
+  | "v3MigratorAddress";
+
+type Networks = {
+  readonly [key in NetworkName]: {
+    readonly [key in ContractType]: string;
+  };
+};
+
+// Maybe this is available somewhere in uniswaps module, but not sure
+const networks: Networks = {
```

Comment:
we should coordinate on _where_ we want to put these kinds of addresses. Ideally, we should have a file within `common` that maps all external addresses used through the repo. For example, right now the trader needs to know about `uniswap v2` router and finder and the new trader (for univ3) needs similar info with the `swapRouter` in particular. We can sync up on this later in the eng meeting.